worker/src/lib.rs

```diff
@@ -0,0 +1,615 @@
+//!struct Features<'a> {
+//!    ///host feature allowing to schedule some work
+//!    schedule: Schedule<'a>,
+//!}
+//!
+//!//custom datatype
+//!struct WorkMessage {
+//!    cycle: usize,
+//!    task: usize,
+//!}
+//!
+//!/// A plugin that do some work in another thread
+//!struct EgWorker {
+//!    //schedule handler need to know the plugin type to use it use WorkData associated type.
+//!    schedule_handler: ScheduleHandler<Self>,
+//!    cycle: usize,
+//!    end_cycle: usize,
+//!}
+//!
+//!/// URI identifier
+//!unsafe impl UriBound for EgWorker {
+//!    const URI: &'static [u8] = b"urn:rust-lv2-more-examples:eg-worker-rs\0";
+//!}
+//!
+//!impl Plugin for EgWorker {
+//!    type Ports = Ports;
+//!    type Features = Features<'static>;
+//!
+//!    fn new(_plugin_info: &PluginInfo, features: Features<'static>) -> Option<Self> {
+//!        // build the schedule_handler using the schedule host feature.
+//!        let schedule_handler = ScheduleHandler::from(features.schedule);
+//!        Some(Self {
+//!            schedule_handler,
+//!            cycle: 0,
+//!            end_cycle: 1,
+//!        })
+//!    }
+//!
+//!    fn run(&mut self, _ports: &mut Ports) {
+//!        self.cycle += 1;
+//!        let cycle = self.cycle;
+//!        println!("cycle {} started", cycle);
+//!        for task in 0..10 {
+//!            let work = WorkMessage { cycle, task };
+//!            // schedule some work and passing some data
+//!            let _ = self.schedule_handler.schedule_work(work);
+//!        }
+//!    }
+//!
+//!    fn extension_data(uri: &Uri) -> Option<&'static dyn Any> {
+//!        match_extensions![uri, WorkerDescriptor<Self>]
+//!    }
+//!}
+//!
+//!/// Implementing the extension.
+//!impl Worker for EgWorker {
+//!    // data type sent by the schedule handler and received by the `work` method.
+//!    type WorkData = WorkMessage;
+//!    // data type sent by the response handler and received by the `work_response` method.
+//!    type ResponseData = String;
+//!    fn work(
+//!        &mut self,
+//!        //response handler need to know the plugin type.
+//!        response_handler: &ResponseHandler<Self>,
+//!        data: Self::WorkData,
+//!    ) -> Result<(), WorkerError> {
+//!        println!("work received: cycle {}, task {}", data.cycle, data.task);
+//!        if data.task >= 5 {
+//!            let _ = response_handler.respond(format!( "response to cycle {}, task {}",
+//!                data.cycle, data.task
+//!            ));
+//!        };
+//!        Ok(())
+//!    }
+//!
+//!    fn work_response(&mut self, data: Self::ResponseData) -> Result<(), WorkerError> {
+//!        println!("work_response received: {}", data);
+//!        Ok(())
+//!    }
+//!
+//!    fn end_run(&mut self) -> Result<(), WorkerError> {
+//!        println!("cycle {} ended", self.end_cycle);
+//!        self.end_cycle += 1;
+//!        Ok(())
+//!    }
+//!}
+//!
+//!lv2_descriptors!(EgWorker);
```

Comment:
What is the problem here ? What manipulation produce an error ? `cargo check --examples` doesn't produce any error here.

Reply:
When you open a shell in `worker` and run `cargo test --all-features`, this test fails. It isn't covered by Travis CI yet because the worker crate isn't included in the workspace yet.